ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/discovery/internal/RecursivePeerRefreshState.java

```diff
@@ -390,4 +379,9 @@ public void cancelTimeout() {
       timeoutCancelled.set(true);
     }
   }
+
+  @FunctionalInterface
+  public interface OutboundDiscoveryMessagingPermissions {
+    boolean isPermitted(Peer remotePeer);
```

Comment:
Maybe `isRemotePermitted`?  In case we add this interface to an uber-permissions object with inbound permissions implicit on the local peer, so the names won't collide?

Reply:
I've got a follow-up PR that will rework these interfaces.  Right now I'm moving towards an API like:  isPermitted(localNode, remotePeer, action).